experimental/personnames/src/derive_name_order.rs

```diff
@@ -0,0 +1,111 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_locid::subtags::Language;
+use icu_locid::Locale;
+use icu_locid_transform::fallback::LocaleFallbacker;
+use zerovec::VarZeroVec;
+
+use crate::api::FormattingOrder;
+
+///
+/// https://www.unicode.org/reports/tr35/tr35-personNames.html#derive-the-name-order
+///
+pub(crate) fn name_order_derive(
+    person_name_locale: &Locale,
+    surname_first: &VarZeroVec<str>,
+    given_first: &VarZeroVec<str>,
+) -> FormattingOrder {
+    // Set up a LocaleFallbacker with data.
+    let fallbacker = LocaleFallbacker::new();
```

Comment:
We should think a bit harder about how to implement this function with pluggable data